test/plots/heatmap.js

```diff
@@ -0,0 +1,76 @@
+import * as Plot from "@observablehq/plot";
+import * as d3 from "d3";
+
+export default async function () {
+  return Plot.plot({
+    color: {
+      type: "diverging"
+    },
+    marks: [
+      Plot.raster({
+        fill: (x, y) => x * y * Math.sin(x) * Math.cos(y),
+        x1: 0,
+        y1: 0,
+        x2: 4 * Math.PI,
+        y2: 4 * Math.PI * (350 / 580),
+        pixelRatio: 3
+      }),
+      Plot.frame()
+    ]
+  });
+}
+
+export async function heatmapArray() {
+  const x1 = 0;
+  const y1 = 0;
+  const x2 = 4 * Math.PI;
+  const y2 = 4 * Math.PI * (350 / 580);
+  const pixelRatio = 3;
+  const width = Math.round(580 / pixelRatio);
+  const height = Math.round(350 / pixelRatio);
+  const x = (f => i => f.invert(i % width + 0.5))(d3.scaleLinear([x1, x2], [0, width])); // prettier-ignore
+  const y = (f => i => f.invert(Math.floor(i / width) + 0.5))(d3.scaleLinear([y2, y1], [height, 0])); // prettier-ignore
+  return Plot.plot({
+    color: {
+      type: "diverging"
+    },
+    marks: [
+      Plot.raster(d3.range(width * height), {
```

Comment:
would this be easier if the signature for fill was `f({x,y}, i)`?

Reply:
I don’t think this test needs to be “easy”—the easy form is shown above, and this test isn’t supposed to be representative of recommended usage. The point here is to test the explicit form where _fill_ is a function of data.